pkg/authorization/registry/resourceaccessreview/rest.go

```diff
@@ -32,29 +33,53 @@ func (r *REST) New() runtime.Object {
 func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
 	resourceAccessReview, ok := obj.(*authorizationapi.ResourceAccessReview)
 	if !ok {
-		return nil, errors.NewBadRequest(fmt.Sprintf("not a resourceAccessReview: %#v", obj))
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a resourceAccessReview: %#v", obj))
 	}
 	if err := kutilerrors.NewAggregate(authorizationvalidation.ValidateResourceAccessReview(resourceAccessReview)); err != nil {
 		return nil, err
 	}
-
-	namespace := kapi.NamespaceValue(ctx)
-
-	attributes := &authorizer.DefaultAuthorizationAttributes{
-		Verb:     resourceAccessReview.Verb,
-		Resource: resourceAccessReview.Resource,
+	// if a namespace is present on the request, then the namespace on the SAR must be empty.  In such a case,
+	// the namespace on the RAR is overwritten.  This is to support backwards compatibility.  To have gotten here
+	// in this state, it means that the authorizer decided that a user could run an RAR against this namespace
+	if namespace := kapi.NamespaceValue(ctx); len(namespace) > 0 {
+		resourceAccessReview.Action.Namespace = namespace
+	}
+	if err := r.isAllowed(ctx, resourceAccessReview); err != nil {
+		return nil, err
 	}
 
-	users, groups, err := r.authorizer.GetAllowedSubjects(ctx, attributes)
+	requestContext := kapi.WithNamespace(ctx, resourceAccessReview.Action.Namespace)
+	attributes := authorizer.ToDefaultAuthorizationAttributes(resourceAccessReview.Action)
+	users, groups, err := r.authorizer.GetAllowedSubjects(requestContext, attributes)
 	if err != nil {
 		return nil, err
 	}
 
 	response := &authorizationapi.ResourceAccessReviewResponse{
-		Namespace: namespace,
+		Namespace: resourceAccessReview.Action.Namespace,
 		Users:     users,
 		Groups:    groups,
 	}
 
 	return response, nil
 }
+
+// isAllowed checks to see if the current user has rights to issue a LocalSubjectAccessReview on the namespace they're attempting to access
+func (r *REST) isAllowed(ctx kapi.Context, rar *authorizationapi.ResourceAccessReview) error {
```

Comment:
> should this check only be done when a namespace is present? checking for permission to run a localresourceaccessreview at the cluster scope is weird

Weird, but essential if we ever completely remove local access reviews.  I think I talked @smarterclayton out of it, but I'm not completely certain.

Also, for people who believe that "" means "all namespaces", the check makes perfect sense.  To check across all namespaces, you must be able to local access reviews against all namespaces.